### ISSUE TITLE
Resolving reverse DNS is slow

### ISSUE THREAD

**Issue by cobarx** [PeelTechnologies/react-native-tcp#39]
On some devices (original Moto G, Galaxy S4) the reverse DNS lookup that takes place in TcpSockets.onConnect is exceedingly slow, taking about 10 seconds. Can this be changed to skip the reverse DNS lookup and always return an IP address?

Unless there is some reason that it needs to return the host name, the following change should handle the issue.
`addressParams.putString("address", address.getHostName());`
to
`addressParams.putString("address", address.getHostString());`

**Comment by cobarx** [comment]
I spoke too soon and didn't realize getHostString only works on API 19 and above.

The following appears to work fine for both hostnames and IPv4 addresses. I did not test it with an IPv6 address:
```
// Calling getHostName() can force a slow reverse DNS lookup, so calculate the IP without a lookup
int ipStartPosition = address.getAddress().toString().indexOf("/") + 1;
addressParams.putString("address", address.getAddress().toString().substring(ipStartPosition));
```

This will need to be applied in both onConnection and onConnect. Let me know if you want me to create a PR for this.

**Comment by aprock** [comment]
should be fixed by #41 
@sivaatpeel can you merge?
